addons/stock_account/models/stock.py

```diff
@@ -418,6 +418,16 @@ def _action_done(self):
             move._run_valuation()
         for move in res.filtered(lambda m: m.product_id.valuation == 'real_time' and (m._is_in() or m._is_out() or m._is_dropshipped() or m._is_dropshipped_returned())):
             move._account_entry_move()
+
+        products_to_vacuum = self.env['product.product']
+        for move in res.filtered(lambda m: m.product_id.valuation == 'real_time' and m._is_in() and (m.product_id.property_cost_method == 'fifo' or m.product_id.categ_id.property_cost_method == 'fifo')):
+            products_to_vacuum += move.product_id
+        if products_to_vacuum:
+            company = res.mapped('company_id') and res.mapped('company_id')[0] or self.env.company
```

Comment:
It would be better to group move by their company instead of taking the first of all move.

Reply:
and do one call to vacuum by company

Reply:
@amoyaux Thanks for the review.

I've grouped products by company linked the move.